### DISCUSSION TITLE
How to interact with UI through gesture rays ？

### DISCUSSION THREAD

**Discussion by jike2018** [StereoKit/StereoKit#940] [Q&A]
How to interact with UI through gesture rays ？

**Comment by maluoi** [comment]
You may want to try out the v0.4.0 preview builds! One major feature in v0.4.0 is UI support for far interaction rays, which is on by default. I'm still working on it, so it may change yet, but it's pretty stable and usable at the moment!

**Reply by jike2018** [reply]
When will version 4.0 be released? I am looking forward to it very much

**Reply by maluoi** [reply]
You can use it now! You just need to pick [one of the preview builds](https://www.nuget.org/packages/StereoKit/#versions-body-tab). The preview builds may be previews, but they tend to be pretty stable. I try not to change StereoKit's APIs much unless it's really necessary, and while v0.4 is my opportunity to change a bunch, I'm still going to change as little as I can.

I don't have a specific date for v0.4.0 yet and it could be a while before I finish it? So if you require the far interaction feature, I'd just start using it now!
